diff --git a/cluster/update-storage-objects.sh b/cluster/update-storage-objects.sh
index 364c1d7..c1b214a 100755
--- a/cluster/update-storage-objects.sh
+++ b/cluster/update-storage-objects.sh
@@ -106,7 +106,7 @@ do
           echo "Looks like ${instance} got deleted. Ignoring it"
           continue
         fi
-        output=$("${KUBECTL}" replace -f "${filename}" --namespace="${namespace}") || true
+        output=$(kput -f "${filename}" --namespace="${namespace}") || true
         rm "${filename}"
         if [ -n "${output:-}" ]
         then
